src/profile/webid.test.ts

```diff
@@ -746,3 +750,34 @@ describe("getPodUrlAllFrom", () => {
     ).toStrictEqual([MOCK_STORAGE, ALT_MOCK_STORAGE]);
   });
 });
+
+describe("getWebIdDataset", () => {
+  it("returns a Solid Dataset for a given WebID", async () => {
+    const webIdProfile = mockProfileDoc("https://some.profile", MOCK_WEBID, {});
+    const { fetch: mockedUnauthFetch } = jest.requireMock("cross-fetch") as {
+      fetch: jest.Mock<
+        ReturnType<typeof window.fetch>,
+        [RequestInfo | URL, RequestInit?]
+      >;
+    };
+    mockedUnauthFetch.mockResolvedValueOnce(
+      new Response(await triplesToTurtle(toRdfJsQuads(webIdProfile)), {
+        headers: {
+          "Content-Type": "text/turtle",
+        },
+      })
+    );
+    const result = await getWebIdDataset(MOCK_WEBID);
+    expect(result?.graphs).toEqual(webIdProfile.graphs);
+  });
+  it("throws an error if fetching fails", async () => {
+    const { fetch: mockedUnauthFetch } = jest.requireMock("cross-fetch") as {
+      fetch: jest.Mock<
+        ReturnType<typeof window.fetch>,
+        [RequestInfo | URL, RequestInit?]
+      >;
+    };
+    mockedUnauthFetch.mockRejectedValueOnce(new Error("error"));
+    await expect(getWebIdDataset(MOCK_WEBID)).rejects.toThrow("error");
```

Comment:
Might be good to make the error message something like `Fetch Failed 500 internal server error` just so the code can't be satisfied by just doing `new Error("error")` or something — i.e., swallowing the error